### ISSUE TITLE
[Issue]: Generate immediately auto-aborts sometimes

### ISSUE THREAD

**Issue by exodia52** [vladmandic/automatic#208]
### Issue Description

This is getting out of hand. The **Generate** Button freezing getting worst and worst. Sometimes even just 1 **Generate** already Freeze it. It happens randomly. Even when "Generating Forever" it stop by itself because of the Frozen Generate button. Couldn't Generate anymore but able to go to other "Tabs" like "Settings", "Img2Img", "Trains" etc.

Every time the **Generate** button freeze it will always tries to generate an image then shows the "Progress Bar" then fully complete but no output was shown or done.

Setup.log doesn't show any error. Same goes for the Console. It just refuse to be pressed until you Refresh the WebUI and sometimes it will let you Generate 1 image then froze again.

### Version Platform Description

Version: 09a14802 Mon Apr 17 15:52:26 2023 -0400

Window 10, Ryzen 7 2700, RTX 3060 Ti

**Comment by vladmandic** [comment]
I'll take a look, but its really hard without the way to reproduce.

**Comment by exodia52** [comment]
The Generate Button freezed again. After 8 Generated Image.

**Comment by vladmandic** [comment]
when this happens, is progess bar still displayed? is it marked as at 100%?

**Comment by exodia52** [comment]
It shows for awhile. Says "Waiting" then it died and disappear. Like nothing happened.

It just as the video shows. At 0:05s.

**Comment by mart-hill** [comment]
I can confirm that it happened to me with older UI commit (shortly after it was reported in #190 [the Generate button didn't do anything and couldn't do anything anymore with it, I had to restart the UI; the rest of the UI was responsive though, I didn't check, if the queue is locked out though]). Could really fast double-click, or accidental right- then left-click cause that behavior?

**Comment by vladmandic** [comment]
i have an idea, can you try something? ui -> settings -> live preview -> Progressbar/preview update period
default is 1000, set it to some really low value like 50

**Comment by exodia52** [comment]
> > Hmm.. After the latest commit i cannot even Generate or Restart the UI at the Extension or at the Settings.
> 
> which commit hash? i've inadvertently created a bug that breaks generate completely, but fixed it 10min later. you should be on `e8d8dae4c71f3bc5043e972620c37921cdf7def3`

No wonder. My bad also. I didn't update my folder. . . Even tho i saw already there's a new fix in commit.

**Comment by vladmandic** [comment]
> Trying the above solution after my console finish loading

its not a solution, its an idea i have on what's the root cause. if that changes behavior, then i know what the issue is and fix will come shortly.

**Comment by exodia52** [comment]
Alrighty, Testing my trusty Generate Forever to catch that bugger. Will update when it freeze again. Usually won't take long. Hopefully.

**Comment by mart-hill** [comment]
I've noticed interesting thing:
![image](https://user-images.githubusercontent.com/55754730/232821292-60fe1415-87b1-4a3a-b7d3-db148075830e.png)

That's despite having negative prompt (two negative embeddings names included). The both numbers are white, when the prompt fields are empty. But either after generation, or after a while, the negative part stops parsing. Could that be something?

**Comment by mart-hill** [comment]
Thank you, that's a BIG benefit! 🙂

**Comment by exodia52** [comment]
Thanks the update !

For my Generate button freezing sometimes it happens after 20+ Generate. Which is better but the problem still floating.

I been playing whack a mole game with another problem pop out. Tell me if you want me to open another issue thread instead of dumping it here.

Traceback (most recent call last):
  File "C:\Users\zerok\Downloads\Test\automatic2\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "C:\Users\zerok\Downloads\Test\automatic2\venv\lib\site-packages\gradio\blocks.py", line 1078, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "C:\Users\zerok\Downloads\Test\automatic2\venv\lib\site-packages\gradio\blocks.py", line 994, in postprocess_data
    block = self.blocks[output_id]
KeyError: 373

This happened when restarting the UI at Extension setting. Which make the UI freeze. The Error appear when clicking the "

----------------------

Another problem when Console starting there's an issue with Diffusers downgrade to 0.14.0 instead of 0.15.0 as stated in the Requirement.txt.

**Comment by vladmandic** [comment]
for this issue, lets stay on topic of frozen generate button. and to clarify, when this happens:
progress bar is not visible, generate is visible, just when you press generate nothing happens?
is there anything in a) console, b) browser console

regarding keyerror, on-the-fly restart of gradio is pretty much a best effort - there is no way to get to 100% unless you restart server from command line. so lets not add that to this issue.

regarding diffusors - can you clarify? current requirements state `diffusers==0.15.0` - do you want to use 0.14.0 (and why?) or is it opposite, you want to use 0.15, but something is downgrading them? if its latter, its 99% some extension you've installed that has different requirements (for example, dreambooth extension is notorious for doing that) and forces installation of its requirements upon load.

**Comment by exodia52** [comment]
> for this issue, lets stay on topic of frozen generate button. and to clarify, when this happens: progress bar is not visible, generate is visible, just when you press generate nothing happens? is there anything in a) console, b) browser console

Alright will stay on the topic.

The progress bar does show up but it really fast now after we set the Preview to `ui -> settings -> live preview -> Progressbar/preview update period to 50`.

The console does not try to "Generate" the picture even tho the Button has been pressed but the button does Grey out where you can "Stop/Skip" for few sec and turn back to Orange "Generate" button.

Sadly this the worst part. There's no Error in the Console.
Where can i find this "Browser Console" ? If you meant by notification pop-out on the top right there's no error.

Even tried the `--disable-queue` and had no success for this issue.

Do note that i'm using Opera GX but i always been using this Browser with other WebUI and had no issue.

> regarding diffusors - can you clarify? current requirements state `diffusers==0.15.0` - do you want to use 0.14.0 (and why?) or is it opposite, you want to use 0.15, but something is downgrading them? if its latter, its 99% some extension you've installed that has different requirements (for example, dreambooth extension is notorious for doing that) and forces installation of its requirements upon load.

I do have Dreambooth extension. Probably because of that. It just being notorious on downgrading mine while i keep trying to upgrade it to prevent any issue. Will noted on this and remove Dreambooth for now to avoid any issue.

Will skip this topic for now till i can pin down the other issue.

**Comment by vladmandic** [comment]
> Where can i find this "Browser Console" ? If you meant by notification pop-out on the top right there's no error.

pretty much every browser has browser console. on edge/chrome, its inside of "developer tools" or sometimes its called "inspector". again, on edge/chrome, you can show it via menu or by pressing ctrl+shift+c. i have no idea about opera.

> The console does not try to "Generate" the picture even tho the Button has been pressed but the button does Grey out where you can see "Stop/Skip" for few sec and turn back to Orange "Generate" button.

thats not how it looks in default `black-orange` theme. theme should not matter, but can you check just so we get closer to nailing this down.

> The progress bar does show up but it really fast now

so progress bar shows up, but then dissapears and nothing gets generated? that sounds like it started the task, but then aborted it.

**Comment by vladmandic** [comment]
so basically, no errors in browser console as well (it would show up already).
i need to track down why the task gets aborted, non-trivial. i wish i could reproduce, that would make it sooo much simpler.

**Comment by exodia52** [comment]
Been generating over 100 images now. Felt like it because of the Theme ?

**Comment by vladmandic** [comment]
theme should really not matter, i just asked to switch to default so we're talking about same visuals (whats visible, whats not, etc)

**Comment by exodia52** [comment]
Then i'll be Generating Forever again. Seems like the issue didn't appear. Which is quite rare for me to get more than 100 Generated image while not getting any Freeze.

Will update in the morning.

**Comment by exodia52** [comment]
Been generating for more than 6 hr+ and no issue. Generated more than 400 Image.

I changed the `ui -> settings -> live preview -> Progressbar/preview update period to 100.` bcs 50 were too fast.
Server arguments: disable-queue,--medvram,--no-half-vae
Deleted Dreambooth extension.
Off `Show previews of all images generated in a batch as a grid` since i don't need it.

That's the only notable thing i did is above and for now no issue.

**Comment by vladmandic** [comment]
50ms update period is definitely not needed, that was just for me to confirm my theory on the root cause.
now that progress check is done differently, even very high update period like 5000ms (5sec) should not cause issues,
but i think 250ms is a good default. faster than that puts a lot more stress on the browser so if you don't have high end pc, it can cause slowdowns.

anyhow, glad its working now. closing of now, but if it reoccurs, i can reopen the issue.

**Comment by exodia52** [comment]
> 50ms update period is definitely not needed, that was just for me to confirm my theory on the root cause. now that progress check is done differently, even very high update period like 5000ms (5sec) should not cause issues, but i think 250ms is a good default. faster than that puts a lot more stress on the browser so if you don't have high end pc, it can cause slowdowns.
> 
> anyhow, glad its working now. closing of now, but if it reoccurs, i can reopen the issue.

I will try the 250ms when i get back from work. Sometimes on the 50ms or 100ms i get memory ran out and the browser crashed. Not a bit deal since simple refresh, easily able to resume what i been doing.

Thank you so much for the active helps. Really appreciate what you been doing. Have a great day.

**Comment by merodiro** [comment]
I have the same issue. This is the browser console output.
it works for some time when I refresh my browser without restarting the server

![image](https://user-images.githubusercontent.com/17033502/236515916-ade04605-e13e-43a6-84ab-994a88a948e0.png)

**Comment by vladmandic** [comment]
doesn't look the same, can you create a new issue for this?

**Comment by mart-hill** [comment]
For the time being, I can use the UIs (@vladmandic's fork and the @anapnoe's fork) thanks to the --disable-queue parameter. Freezing is inevitable for me, no matter if I use xformers or SDP optimization. The other way to try to avoid it would be to "be adventurous" and try torch 2.1 nightly, as @vladmandic suggested. 🙂 
I'm really curious, as to why websockets communication seems to be at least the catalyst for this freeze to happen...

**Comment by anapnoe** [comment]
> For the time being, I can use the UIs (@vladmandic's fork and the @anapnoe's fork) thanks to the --disable-queue parameter. Freezing is inevitable for me, no matter if I use xformers or SDP optimization. The other way to try to avoid it would be to "be adventurous" and try torch 2.1 nightly, as @vladmandic suggested. 🙂 I'm really curious, as to why websockets communication seems to be at least the catalyst for this freeze to happen...

can you try to generate a small image like 32x32 px?
for me it seems like there is some kind of overflow that crash the script engine
if you disable the preview does it work?

**Comment by mart-hill** [comment]
I had the Generate button freeze even without a previews, so this option is the only way for me. It's exactly as @exodia52 showed in his video up here. 
I'll try to generate such small image with your fork, since it freezes earlier on me. Of course it won't, when I use --disable-queue parameter, so I'll run it without it.

Could that be that the `shared.demo.queue(64)` in your fork's webui.py vs @vladmandic's `shared.demo.queue(16)` matters somehow?

**Comment by anapnoe** [comment]
This parameter is used to set the number of worker threads in the Gradio server that will be processing requests in parallel. 
so maybe 64 it to high I haven't tested

**Comment by vladmandic** [comment]
It was 16 in my fork until two days ago, so that's not it.
